simphony_mayavi/plugins/engine_manager_mayavi2.py

```diff
@@ -0,0 +1,128 @@
+import logging
+
+from traits.api import Instance, List, on_trait_change
+from traitsui.api import View, VGroup, Group, Item
+from envisage.api import Plugin, ServiceOffer
+
+# EngineManager imports mayavi.core.trait_defs in the module level
+from simphony_mayavi.plugins.engine_manager import EngineManager
+from simphony_mayavi.plugins.tabbed_panel_collection import (
+    TabbedPanelCollection)
+
+# More internal mayavi imports are done within EngineManagerMayavi2 to
+# avoid import cycles when the class is imported as a Mayavi2 plugin
+
+logger = logging.getLogger(__name__)
+
+
+class EngineManagerMayavi2(EngineManager):
+
+    window = Instance("pyface.workbench.api.WorkbenchWindow")
+    panels = Instance(TabbedPanelCollection)
+
+    view = View(
+        VGroup(
+            Group(Item("engine_name", label="Engine Wrapper")),
+            Group(Item("panels", style="custom", show_label=False))),
+        resizable=True)
+
+    def get_mayavi(self):
+        from mayavi.core.engine import Engine
+        return self.window.get_service(Engine)
+
+    def _window_changed(self):
+        self._set_panels()
+
+    def _set_panels(self):
+        # mayavi imports here to avoid import cycle
+        from simphony_mayavi.plugins.add_source_panel import AddSourcePanel
+        from simphony_mayavi.plugins.run_and_animate_panel import (
+            RunAndAnimatePanel)
+
+        mayavi_engine = self.get_mayavi()
+
+        # Add panels
+        self.panels = TabbedPanelCollection(
+            add_source=AddSourcePanel(self.engine_name, self.engine,
+                                      mayavi_engine),
+            run_and_animate=RunAndAnimatePanel(self.engine,
+                                               mayavi_engine))
+        logger.info("Simphony EngineManagerMayavi2 panel setup completed")
+
+    def _engine_name_changed(self):
+        for panel in self.panels:
+            panel.engine = self.engine
+        self.panels.add_source.engine_name = self.engine_name
+
+
+class EngineManagerMayavi2Plugin(Plugin):
+
+    SERVICE_OFFERS = "envisage.ui.workbench.service_offers"
+    VIEWS = "envisage.ui.workbench.views"
+
+    service_offers = List(contributes_to=SERVICE_OFFERS)
+    views = List(contributes_to=VIEWS)
+
+    # Private methods.
+    def _service_offers_default(self):
+        """ Trait initializer. """
+        worker_service_offer = ServiceOffer(
+            protocol=('simphony_mayavi.plugins.engine_manager_mayavi2.'
```

Comment:
it's better to keep the protocol name as one single string (even if too long) and add a `  # noqa`. If not, searching for that string will always fail on those entries.